### PR TITLE
🌱 syncer vw: add API definition logging

### PR DESCRIPTION
**Pull request by ncdc**
Trying to figure out #2762

### CODE REVIEW

File: pkg/virtual/syncer/controllers/apireconciler/syncer_apireconciler_indexes.go [hunk 1]
```diff
@@ -43,23 +38,20 @@ func IndexAPIExportsByAPIResourceSchemas(obj interface{}) ([]string, error) {
...
-	clusterName := logicalcluster.From(synctarget)
-	if len(synctarget.Spec.SupportedAPIExports) == 0 {
+	clusterName := logicalcluster.From(syncTarget)
+	if len(syncTarget.Spec.SupportedAPIExports) == 0 {
 		return []string{client.ToClusterAwareKey(clusterName.Path(), reconcilerapiexport.TemporaryComputeServiceExportName)}, nil
 	}
 
-	keys := make([]string, 0, len(synctarget.Spec.SupportedAPIExports))
-	for _, export := range synctarget.Spec.SupportedAPIExports {
-		if len(export.Export) == 0 {
```
Comment (ncdc) [inline pkg/virtual/syncer/controllers/apireconciler/syncer_apireconciler_indexes.go:58]: FYI @davidfestal @sttts I'm pretty sure this is a bug, right?

Comment (sttts) [reply pkg/virtual/syncer/controllers/apireconciler/syncer_apireconciler_indexes.go:58]: looks like, yes

Comment (ncdc) [reply pkg/virtual/syncer/controllers/apireconciler/syncer_apireconciler_indexes.go:58]: There is a _chance_ this is the root cause of #2762

Comment (davidfestal) [reply pkg/virtual/syncer/controllers/apireconciler/syncer_apireconciler_indexes.go:58]: I agree it seems a bug that might be related to the mentioned issue.

**Pull request merged**